eng/common/scripts/Package-Properties.ps1

```diff
@@ -105,6 +105,38 @@ function Get-PkgProperties
     return $null
 }
 
+function Get-PackageFolderFromPath($file) {
+    $parts = $file -split "/"
+
+    if ($parts.Length -lt 3) {
+        return ""
+    }
+    return [System.IO.Path]::Combine($parts[0], $parts[1], $parts[2])
+}
+
+# Takes an input diff json file generated from eng/common/scripts/Generate-PR-Diff.ps1
+# On PR builds, this function should be utilized instead of Get-AllPkgProperties.
+function Get-PrPkgProperties([string]$InputDiffJson) {
+    $packagesWithChanges = @()
+
+    $allPackageProperties = Get-AllPkgProperties
+    $diff = Get-Content $InputDiffJson | ConvertFrom-Json
+    $targetedFiles = $diff.ChangedFiles
+
+    foreach($pkg in $allPackageProperties)
+    {
+        foreach($file in $targetedFiles)
+        {
+            $pkgFolder = Get-PackageFolderFromPath($file)
```

Comment:
Should we just check to see if the file is under $pkg.DirectoryPath?

Reply:
Ok.

Reply:
I used `-like` with a path resolution to ensure that the path separators would be consistent.

I could use the same, get the string, and `StartsWith`, but I didn't see that it made much of a difference.